### ISSUE TITLE
Getting exception of node pool when trying to run [acm-terraform-blog-part1]

### ISSUE THREAD

**Issue by arvpai** [terraform-google-modules/terraform-google-kubernetes-engine#1286]
### TL;DR

Error: error creating NodePool: googleapi: Error 400: Creation of node pools using node images based on Docker container runtimes is not supported in GKE v1.23. This is to prepare for the removal of Dockershim in Kubernetes v1.24. We recommend that you migrate to image types based on Containerd (examples). For more information, contact Cloud Support., badRequest .
 with module.gke.google_container_node_pool.pools["default-node-pool"],
│   on .terraform/modules/gke/cluster.tf line 194, in resource "google_container_node_pool" "pools":
│  194: resource "google_container_node_pool" "pools" 

### Expected behavior

Expect a gke cluster as well as ACM being configured after that GKE Cluster is registered with Anthos . Since this is blog 1 we expect config-sync to be set up for ACM.

### Observed behavior

However the terraform apply fails with the above stated expection since in the cluster.tf of the source folder it goes back to the GKE v 1.23 everytime and it does not work out.

Kindly fix this issue since the above examples for blogs 1,2 & 3 will not work.

Snippet of Terraform plan which depicts the older version :

 # module.gke.google_container_cluster.primary will be created
  + resource "google_container_cluster" "primary" {
      + cluster_ipv4_cidr           = (known after apply)
      + datapath_provider           = (known after apply)
      + default_max_pods_per_node   = 110
      + enable_binary_authorization = false
      + enable_intranode_visibility = (known after apply)
      + enable_kubernetes_alpha     = false
      + enable_legacy_abac          = false
      + enable_shielded_nodes       = true
      + endpoint                    = (known after apply)
      + id                          = (known after apply)
      + instance_group_urls         = (known after apply)
      + label_fingerprint           = (known after apply)
      + location                    = "us-central1"
      + logging_service             = "logging.googleapis.com/kubernetes"
      + master_version              = (known after apply)
      + min_master_version          = "1.23.6-gke.1700"
      + monitoring_service          = "monitoring.googleapis.com/kubernetes"
      + name                        = "gke-cluster"

### Terraform Configuration

```hcl
module "enabled_google_apis" {
  source  = "terraform-google-modules/project-factory/google//modules/project_services"
  version = "~> 11.3"

  project_id                  = var.project
  disable_services_on_destroy = false

  activate_apis = [
    "compute.googleapis.com",
    "container.googleapis.com",
    "gkehub.googleapis.com",
    "anthosconfigmanagement.googleapis.com"
  ]
}

module "gke" {
  source             = "terraform-google-modules/kubernetes-engine/google"
  version            = "~> 16.0"
  project_id         = module.enabled_google_apis.project_id
  name               = "sfl-acm-part1"
  region             = var.region
  zones              = [var.zone]
  initial_node_count = 4
  network            = "default"
  subnetwork         = "default"
  ip_range_pods      = ""
  ip_range_services  = ""
}
```

### Terraform Version

```sh
version = "3.73.0"
```

**Comment by bharathkkb** [comment]
Thanks for the report @arvpai 
We will need to [update this ](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/19f59c4d55e0761fac0e37103502c70b90536800/examples/acm-terraform-blog-part1/terraform/gke.tf#L34)to use some of our newer module releases.

**Comment by abhisek00** [comment]
Hi @bharathkkb I am working on this issue.

**Comment by sephethus** [comment]
The lack of helpful information on how to deal with this problem is astounding. This isn't fixed by anything. That link is not helpful at all.

**Comment by Temikus** [comment]
For others looking for what to do, it's most likely the node auto-provisioning that hasn't been updated, this script should help:
https://github.com/GoogleCloudPlatform/k8s-node-tools/blob/HEAD/migrating-to-containerd/find-nodepools-to-migrate.sh
